exchanges/gateio/gateio_convert.go

```diff
@@ -0,0 +1,411 @@
+package gateio
+
+import (
+	"encoding/json"
+	"reflect"
+	"strconv"
+	"time"
+)
+
+type gateioMilliSecTime int64
+
+// UnmarshalJSON deserializes json, and timestamp information.
+func (a *gateioMilliSecTime) UnmarshalJSON(data []byte) error {
+	var value interface{}
+	err := json.Unmarshal(data, &value)
+	if err != nil {
+		return err
+	}
+	val := reflect.ValueOf(value)
+	switch val.Kind() {
+	case reflect.Int64, reflect.Int32:
+		valueInteger, ok := value.(int64)
+		if !ok {
+			valueInteger = 0
+		}
+		*a = gateioMilliSecTime(valueInteger)
+	case reflect.Float64, reflect.Float32:
+		parsedValue, ok := value.(float64)
+		if !ok {
+			parsedValue = 0
+		}
+		*a = gateioMilliSecTime(int64(parsedValue))
+	default:
+		stringValue, ok := value.(string)
+		if !ok {
+			stringValue = "0"
+		}
+		parsedValue, err := strconv.ParseFloat(stringValue, 64)
+		if err != nil {
+			return err
+		}
+		*a = gateioMilliSecTime(int64(parsedValue))
+	}
+	return nil
+}
+
+// Time represents a time instance.
+func (a *gateioMilliSecTime) Time() time.Time {
+	return time.UnixMilli(int64(*a))
+}
+
+type gateioTime int64
+
+// UnmarshalJSON deserializes json, and timestamp information.
+func (a *gateioTime) UnmarshalJSON(data []byte) error {
+	var value interface{}
+	err := json.Unmarshal(data, &value)
+	if err != nil {
+		return err
+	}
+	val := reflect.ValueOf(value)
+	switch val.Kind() {
+	case reflect.Int64, reflect.Int32:
+		valueInteger, ok := value.(int64)
+		if !ok {
+			valueInteger = 0
+		}
+		*a = gateioTime(valueInteger)
+	case reflect.Float64, reflect.Float32:
+		parsedValue, ok := value.(float64)
+		if !ok {
+			parsedValue = 0
+		}
+		*a = gateioTime(int64(parsedValue))
+	default:
+		stringValue, ok := value.(string)
+		if !ok {
+			stringValue = "0"
+		}
+		parsedValue, err := strconv.ParseFloat(stringValue, 64)
+		if err != nil {
+			return err
+		}
+		*a = gateioTime(int64(parsedValue))
+	}
+	return nil
+}
+
+// Time represents a time instance.
+func (a *gateioTime) Time() time.Time {
+	return time.Unix(int64(*a), 0)
+}
+
+// UnmarshalJSON deserializes json data into CurrencyPairDetail
+func (a *CurrencyPairDetail) UnmarshalJSON(data []byte) error {
+	type Alias CurrencyPairDetail
+	chil := struct {
+		*Alias
+		Fee            string `json:"fee"`
+		MinBaseAmount  string `json:"min_base_amount"`
```

Comment:
Rather than creating an alias for these and other string to float64 checks, could create a `gateioNumericalValue` custom type like in OKX / Bybit, this will cut down on having to require many of these UnmarshalJSON handlers